packages/ui/src/components/InputFields/SmartInputField/use-smart-input-field.ts

```diff
@@ -121,6 +122,12 @@ const useSmartInputField = ({ _defaultType, defaultValue, enabledTypes }: Props)
     setCurrentType(enabledTypeSet.size === 1 ? defaultType : undefined);
   }, [defaultType, enabledTypeSet.size]);
 
+  // CAUTION: For preview use only, enabledTypes and defaultType should not be changed after component mounted
```

Comment:
do we have any flag to make sure this happens in preview only?

Reply:
No necessary.  It won't break anything if the enabled types are mutated. The comments here just indicate this is not a use case for production.  In production, all the sign-in-methods settings come from the./well-known  API, only preview mode have run time sign-in-methods mutate event.

Reply:
move forward